string results_dir
---
bool success